app/utils/SHACL/shacl-validate-ttl-code.js

```diff
@@ -0,0 +1,55 @@
+import { getLocalFileContentAsText } from '../get-local-file-content';
+import SHACLValidator from 'rdf-validate-shacl';
+import { DatasetFactory } from 'rdf-ext';
+import factory from '@rdfjs/dataset';
+import { Parser as ParserN3 } from 'n3';
+
+export async function shaclValidateTtlCode(ttlCode) {
+  if (!ttlCode) {
+    return;
+  }
+
+  const rdf = new DatasetFactory();
+  const shapesTtl = await getAllShapesTtl();
+
+  const shapeQuads = await getQuadsFromTtlCode(shapesTtl);
+  const dataQuads = await getQuadsFromTtlCode(ttlCode);
+
+  const dataDataset = rdf.dataset(dataQuads);
+
+  const validator = new SHACLValidator(shapeQuads, {
+    allowNamedNodeInList: true,
+  });
+  const report = validator.validate(dataDataset);
+
+  return report;
+}
+
+async function getAllShapesTtl() {
+  const allShapes = await Promise.all([
+    getLocalFileContentAsText('/SHACL/V2/repositories.ttl'),
+    getLocalFileContentAsText('/SHACL/V2/field.ttl'),
+    getLocalFileContentAsText('/SHACL/V2/section.ttl'),
+    getLocalFileContentAsText('/SHACL/V2/required-constraint.ttl'),
+    getLocalFileContentAsText('/SHACL/V2/max-characters-constraint.ttl'),
```

Comment:
@aatauil  Im not a fan of listing them here. If we want to validate the ttl code again V1 shapes we cannot do this.

Do we just add a parameter to the function that should be the ttlCode of the shapes? And move the getAllShapesTtl to a util? `getAllV1Shapes(ttlCode, shapesTtl)` & `getAllV2Shapes(ttlCode, shapesTtl)` or a parameter for version just thinking at loud